src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs

```diff
@@ -0,0 +1,161 @@
+
+            string tableName = DataTestUtility.GetUniqueNameForSqlServer("Json_Test");
+            string spName = DataTestUtility.GetUniqueNameForSqlServer("spJson_ReadTest");
+
+            string tableCreate = "CREATE TABLE " + tableName + " (Data json)";
+            string tableInsert = "INSERT INTO " + tableName + " VALUES (@jsonData)";
+            string tableRead = "SELECT * FROM " + tableName;
+            string spCreate = "CREATE PROCEDURE " + spName + "AS " + tableRead;
+
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString))
+            {
+                connection.Open();
+                using (SqlCommand command = connection.CreateCommand())
+                {
+                    //Create Table
+                    command.CommandText = tableCreate;
+                    command.ExecuteNonQuery();
+
+                    //Create SP for reading from json column
+                    command.CommandText = spCreate;
+                    command.ExecuteNonQuery();
+
+                    //Insert sample json data
+                    //This will be used for reading
+                    command.CommandText = tableInsert;
+                    var parameter = new SqlParameter("@jsonData", SqlDbTypeExtensions.Json);
+                    parameter.Value = jsonDataString;
+                    command.Parameters.Add(parameter);
+                    command.ExecuteNonQuery();
+
+                    //Test 1
+                    //Read json value using query
+                    command.CommandText = tableRead;
+                    SqlDataReader reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        string jsonData = reader.GetString(0);
+                        _output.WriteLine(jsonData);
+                        Assert.Equal(jsonDataString,jsonData);
+                    }
+
+                    //Test 2
+                    //Read the column metadata
+                    System.Collections.ObjectModel.ReadOnlyCollection<DbColumn> schema = reader.GetColumnSchema();
+                    foreach (DbColumn column in schema)
+                    {
+                        _output.WriteLine("Column Name is " + column.ColumnName);
+                        _output.WriteLine("Column DataType is " + column?.DataType.ToString());
+                        _output.WriteLine("Column DataTypeName is " + column.DataTypeName);
+                        Assert.Equal("json",column.DataTypeName);
+                    }
+                    reader.Close();
+
+                    //Test 3
+                    //Read json value using SP
+                    using (SqlCommand command2 = connection.CreateCommand())
+                    {
+                        command2.CommandText = spName;
+                        command2.CommandType = CommandType.StoredProcedure;
+                        SqlDataReader reader2 = command2.ExecuteReader();
```

Comment:
You could try wrapping this in using block

Reply:
Added `using` block